File: e2e/selenide-junit5-selenium-grid/src/test/java/provider/ValidationProvider.java
package provider;

import interaction.validation.AccessibilityValidator;
import interaction.validation.LanguageValidator;
import interaction.validation.TaskValidator;
import interaction.validation.TasksValidator;
import lombok.Getter;

@Getter
public class ValidationProvider {
    public TasksValidator tasks = new TasksValidator();
    public TaskValidator task = new TaskValidator();
    public LanguageValidator language = new LanguageValidator();
    public AccessibilityValidator accessibility = new AccessibilityValidator();
}

File: demo-android/app/src/androidTest/java/com/example/demo/e2e/provider/ValidationProvider.kt
package com.example.demo.e2e.provider

import androidx.compose.ui.test.junit4.AndroidComposeTestRule
import com.example.demo.MainActivity
import com.example.demo.e2e.interaction.validation.AccessibilityValidator
import com.example.demo.e2e.interaction.validation.LanguageValidator
import com.example.demo.e2e.interaction.validation.TaskValidator
import com.example.demo.e2e.interaction.validation.TasksValidator

class ValidationProvider(
    rule: AndroidComposeTestRule<*, MainActivity>,
) {
    val tasks: TasksValidator = TasksValidator(rule)
    val task: TaskValidator = TaskValidator(rule)
    val language: LanguageValidator = LanguageValidator(rule)
    val accessibility: AccessibilityValidator = AccessibilityValidator()
}
